src/lib/button/button.ts

```diff
@@ -56,6 +56,28 @@ export class MatButtonCssMatStyler {}
 })
 export class MatRaisedButtonCssMatStyler {}
 
+/**
+ * Directive whose purpose is to add the mat- CSS styling to this selector.
+ * @docs-private
+ */
+@Directive({
+  selector: 'button[mat-stroked-button], a[mat-stroked-button]',
+  host: {'class': 'mat-stroked-button'}
+})
+export class MatStrokedButtonCssMatStyler {}
+
+
+
+/**
+ * Directive whose purpose is to add the mat- CSS styling to this selector.
+ * @docs-private
+ */
+@Directive({
+  selector: 'button[mat-flat-button], a[mat-flat-button]',
+  host: {'class': 'mat-flat-button'}
+})
+export class MatFlatButtonCssMatStyler {}
```

Comment:
These separate styler directives aren't really necessary; now would be a good time to do away with them here. You can have `MatButton` just apply the right class checking the attributes (i.e. adding a new method like `_setButtonVariantClass`)